cmd/state-svc/main.go

```diff
@@ -192,6 +192,15 @@ func runForeground(cfg *config.Instance, an *anaSync.Client, auth *authenticatio
 	signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
 	defer signal.Stop(sig)
 
+	p.RunIfNotAuthority(time.Second*3, svcctl.NewDefaultIPCClient(), func(err error) {
```

Comment:
Given this issue seems to stem from processors being overloaded; is 3 seconds enough? What's the harm in giving this like 30 seconds?

End of the day we want to make sure redundant service processes are shut down. I don't think we care as much that it happens within a certain limited timeframe, other than reproducibility of tests which we should be careful about having that influence our shipped logic.

I guess this begs the question; is this PR effectively only reducing the time it would take for the service to shut down? ie. even without these changes the service would inevitably shut itself down anyway?

Reply:
Once the processes lock up the CPU, it seems that the wait time is mostly irrelevant. So, 3 seconds or 30 won't really improve or harm anything. This redundant mechanism isn't for that use case; It's pretty much just for unforeseen cases where service instances do exist while not established as the authoritative instance (i.e. they, somehow, make it past the sock guard).